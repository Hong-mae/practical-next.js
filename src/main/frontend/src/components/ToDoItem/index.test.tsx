import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import { ToDoItem } from '.';

describe('<ToDoItem />', () => {
  it('renders component coreectly', () => {
    const { container } = render(<ToDoItem label="default value" />);

    const toDoItem = screen.getByText('default value');
    expect(toDoItem).toBeInTheDocument();

    const deleteBtn = screen.getByText('삭제');
    expect(deleteBtn).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('clicks the delete button', () => {
    const handleClick = jest.fn();
    render(<ToDoItem label="default value" onDelete={handleClick} />);

    const deleteBtn = screen.getByText('삭제');
    expect(handleClick).toHaveBeenCalledTimes(0);
    fireEvent.click(deleteBtn);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
