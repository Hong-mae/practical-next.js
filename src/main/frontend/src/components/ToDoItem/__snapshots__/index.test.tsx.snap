// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDoItem /> renders component coreectly 1`] = `
.c2 {
  text-align: center;
  background-color: #ff1744;
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
}

.c2:hover {
  background-color: #f01440;
}

.c2:active {
  box-shadow: inset 5px 5px 10px rgba(0, 0, 0, 0.2);
}

.c3 {
  color: #fff;
  font-size: 16px;
}

.c0 {
  display: flex;
  border-bottom: 1px solid #bdbdbd;
  align-items: center;
  margin: 10px;
  padding: 10px;
}

.c1 {
  flex: 1;
  font-size: 16px;
  margin-right: 20px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      default value
    </div>
    <div
      backgroundcolor="#ff1744"
      class="c2"
      hovercolor="#f01440"
    >
      <div
        class="c3"
      >
        삭제
      </div>
    </div>
  </div>
</div>
`;
